### PR TITLE
docs: add documentation for forkToken

### PR DESCRIPTION
**Pull request by Chumper**
## Changes:

Added documentation

## Context:

I provide the forking GitHub App as well internally in the company.
So I completed the documentation for this workflow

## Documentation (please check one with an [x])

- [X] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/self-hosted-configuration.md [hunk 1]
```diff
@@ -193,6 +193,13 @@ It will also override any settings in `packageRules`.
 ## forkMode
 
 You probably have no need for this option - it is an experimental setting for the Renovate hosted GitHub App.
+However, if this is set to true then Renovate will clone the repository before working on the clone and creating Pull Requests on the parent (GitHub only).
+
+## forkToken
```
Comment (rarkins) [inline docs/usage/self-hosted-configuration.md:198]: Needs to be added to lib/config/definitions.ts too

Comment (Chumper) [reply docs/usage/self-hosted-configuration.md:206]: Done

**Review by viceice — DISMISSED** [review]

File: docs/usage/self-hosted-configuration.md [hunk 2]
```diff
@@ -200,6 +200,13 @@ It will also override any settings in `packageRules`.
...
+
+You probably have no need for this option - it is an experimental setting for the Renovate hosted GitHub App.
+This should be set to a personal access token (GitHub only) when `forkMode` is set to `true`.
```
Comment (HonkingGoose) [inline docs/usage/self-hosted-configuration.md:208]:
```suggestion
This should be set to a Personal Access Token (GitHub only) when `forkMode` is set to `true`.
```

I think the term "Personal Access Token" should be capitalized.

File: docs/usage/self-hosted-configuration.md [hunk 3]
```diff
@@ -200,6 +200,13 @@ It will also override any settings in `packageRules`.
 ## forkMode
 
 You probably have no need for this option - it is an experimental setting for the Renovate hosted GitHub App.
+However, if this is set to true then Renovate will clone the repository before working on the clone and creating Pull Requests on the parent (GitHub only).
```
Comment (HonkingGoose) [inline docs/usage/self-hosted-configuration.md:203]:
```suggestion
However, if this is set to `true` then Renovate will clone the repository before working on the clone and creating Pull Requests on the parent (GitHub only).
```

Let's put `true` in monospaced font.

Also it's really confusing to read: "Renovate will clone the repository before working on the clone"...
I don't know what the behavior is, but the way its worded now is not helping me understand whats happening.

The way I understand Renovate is that it _always_ clones the repo before doing stuff???

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

File: docs/usage/self-hosted-configuration.md (hunk 3 above)
Comment (Chumper) [reply docs/usage/self-hosted-configuration.md:203]: Correct, wording is misleading, I meant fork instead of clone. Let me rephrase.

Comment (HonkingGoose) [reply docs/usage/self-hosted-configuration.md:203]: > Correct, wording is misleading, I meant fork instead of clone. Let me rephrase.

I thought some wires got crossed somewhere. 😄 
I'll let you rephrase this. 😉

Comment (Chumper) [reply docs/usage/self-hosted-configuration.md:203]: I have updated the wording, does it make more sense now?

File: docs/usage/self-hosted-configuration.md [hunk 4]
```diff
@@ -200,6 +200,14 @@ It will also override any settings in `packageRules`.
 ## forkMode
 
 You probably have no need for this option - it is an experimental setting for the Renovate hosted GitHub App.
+However, if this is set to `true` then Renovate will fork the repository into the personal space of the person owning the Personal Access Token.
```
Comment (HonkingGoose) [inline docs/usage/self-hosted-configuration.md:203]:
```suggestion
If this is set to `true` then Renovate will fork the repository into the personal space of the person owning the Personal Access Token.
```

I think we can remove the "However" and still be clear.

**Review by HonkingGoose — CHANGES_REQUESTED** [review]
I've made a suggestion to fix up a line of text.

I'm not clear on what "personal space" refers to. What/where is this "personal space"? Do we _need_ to use this term to understand what the feature does differently compare to the default Renovate bot behavior? I'm thinking that maybe this is too much detail and will only confuse readers? But maybe we do need to explain this concept?

**Comment by Chumper** [comment]
The personal space for me is where all repos under https://github.com/Chumper?tab=repositories are, so all repos that are not under an organization like `renovatebot` here.

Maybe `personal repositories` is a better name.

As a PAT always belongs to a user account on GitHub, there is always a personal space where repositories can be forked into.

**Review by rarkins — APPROVED** [review]
